test/common/network/proxy_protocol_test.cc

```diff
@@ -53,7 +54,11 @@ class ProxyProtocolTest : public testing::Test {
   std::shared_ptr<MockReadFilter> read_filter_;
 };
 
-TEST_F(ProxyProtocolTest, Basic) {
+// Parameterize the listener socket address version.
+INSTANTIATE_TEST_CASE_P(IpVersions, ProxyProtocolTest,
+                        testing::ValuesIn(TestEnvironment::getIpVersionsForTest()));
```

Comment:
Why do we even need a real socket_/listener_ here? Seems that this should be mockable. But, I haven't deep dived. WDYT?